src/VASSAL/counters/BasicPiece.java

```diff
@@ -338,6 +359,26 @@ public void setProperty(Object key, Object val) {
     }
   }
 
+  @Override
+  public Command setPersistentProperty(Object key, Object val) {
+    if (persistentProps == null) {
+      persistentProps = new HashMap<>();
+    }
+    final Object oldVal = persistentProps.get(key);
```

Comment:
This call to Map.get() can be omitted, the Map.put() and Map.remove() calls already return the previous value, e.g.:
``
final Object oldVal = (val == null)
  ? props.remove(key)
  : props.put(key, val);
``

Reply:
Great, changed.